cmd/system-probe/modules/traceroute.go

```diff
@@ -0,0 +1,99 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux || windows
+
+package modules
+
+import (
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/DataDog/datadog-agent/cmd/system-probe/api/module"
+	"github.com/DataDog/datadog-agent/cmd/system-probe/config"
+	sysconfigtypes "github.com/DataDog/datadog-agent/cmd/system-probe/config/types"
+	tracerouteutil "github.com/DataDog/datadog-agent/pkg/networkpath/traceroute"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/gorilla/mux"
+	"go.uber.org/atomic"
+	"google.golang.org/grpc"
+)
+
+type traceroute struct{}
+
+// Traceroute is a factory for NDMs Traceroute module
+var Traceroute = module.Factory{
+	Name:             config.TracerouteModule,
+	ConfigNamespaces: []string{"traceroute"},
+	Fn: func(cfg *sysconfigtypes.Config) (module.Module, error) {
+		return &traceroute{}, nil
+	},
+	NeedsEBPF: func() bool {
+		return false
+	},
+}
+
+var _ module.Module = &traceroute{}
+
+func (t *traceroute) GetStats() map[string]interface{} {
+	return nil
+}
+
+func (t *traceroute) Register(httpMux *module.Router) error {
+	var runCounter = atomic.NewUint64(0)
+
+	httpMux.HandleFunc("/traceroute/{host}", func(w http.ResponseWriter, req *http.Request) {
```

Comment:
We are only passing the destination `host/hostname` here.

What's the plan for passing future options (protocol, port, timeout, etc) ? mostly optional I think.

Reply:
For ping, we pass additional configuration using query parameters but that may or may not make sense here. I think that's a decision best left to a future iteration with the traceroute implementation chosen.